redwood-layout-shared-test/src/commonMain/kotlin/app/cash/redwood/layout/AbstractBoxTest.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.layout
+
+import app.cash.redwood.layout.api.Constraint
+import app.cash.redwood.layout.widget.Box
+import app.cash.redwood.ui.Dp
+import app.cash.redwood.ui.dp
+import kotlin.test.Test
+
+abstract class AbstractBoxTest<T : Any> {
+
+  abstract fun Box(block: Box<T>.() -> Unit = {}): Box<T>
+
+  abstract fun Color(block: Color<T>.() -> Unit = {}): Color<T>
+
+  fun Color(color: Int, width: Dp, height: Dp) = Color {
+    color(color)
+    width(width)
+    height(height)
+  }
+
+  abstract fun verifySnapshot(value: T)
+
+  @Test
+  open fun testDefaults() {
```

Comment:
Huh do the other test classes use `open`? I don't recall this being needed, but maybe it is.

Reply:
That's left over debug code so I could override + run in the IDE. Will update!